src/plugins/dashboard/public/application/lib/sync_dashboard_container_input.ts

```diff
@@ -111,6 +112,10 @@ export const applyContainerChangesToState = ({
     dispatchDashboardStateChange(setQuery(input.query));
   }
 
+  if (input.timeRestore && !_.isEqual(input.timeRange, latestState.timeRange)) {
+    dispatchDashboardStateChange(setTimeRange(input.timeRange));
+  }
```

Comment:
Only compare `timeRange` when `timeRestore == true` - if `timeRestore == false`, it doesn't *matter* if `timeRange` changed on sync (i.e. changes in `timeRange` shouldn't cause unsaved changes unless `timeRestore == true`